akvo/rsr/management/commands/migrate_related_projects.py

```diff
@@ -0,0 +1,223 @@
+# -*- coding: utf-8 -*-
+
+# Akvo Reporting is covered by the GNU Affero General Public License.
+# See more details in the license.txt file located at the root folder of the Akvo RSR module.
+# For additional details on the GNU license please see < http://www.gnu.org/licenses/agpl.html >.
+import dataclasses
+import operator
+from argparse import ArgumentParser
+from functools import reduce
+from typing import Dict, List, Set, Hashable, Generic, TypeVar
+from uuid import UUID
+
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from akvo.rsr.models import Project, RelatedProject
+
+
+class Command(BaseCommand):
+    help = "Create a tree structure of Projects from RelatedProjects"
+
+    def add_arguments(self, parser: ArgumentParser):
+        parser.add_argument(
+            "--apply", action="store_true",
+            help="Actually apply the changes"
+        )
+
+    def handle(self, *args, **options):
+        try:
+            migrate(options.get("apply", False))
+        except InterruptedError:
+            print("Changes not applied")
+        else:
+            print("Changes applied")
+        print("DONE!")
+
+
+@transaction.atomic
+def migrate(apply=False):
+    modified_projects = []
+    # Handle parent child relationships
+    parent_child_related_projects = RelatedProject.objects.filter(
+        relation__in=[RelatedProject.PROJECT_RELATION_PARENT, RelatedProject.PROJECT_RELATION_CHILD],
+        related_project__isnull=False,
+    )
```

Comment:
There might be some old data with circular relationships, we can ignore those data, please handle this edge case so it's not making the command fail when running